include/geometry.tcc

```diff
@@ -0,0 +1,45 @@
+//! Compute inverse of 2d rotation matrix for orthogonal axis coordinate system
+template <>
+inline Eigen::Matrix<double, 2, 2> mpm::Geometry<2>::inverse_rotation_matrix(
+    const Eigen::Matrix<double, 2, 1>& angles) {
+
+  // Get cos and sin of angles
+  const double cos_alpha = cos(angles(0));
+  const double sin_alpha = sin(angles(0));
+  const double cos_beta = cos(angles(1));
+  const double sin_beta = sin(angles(1));
+
+  // clang-format off
+  const Eigen::Matrix<double, 2, 2> rotation_matrix = 
```

Comment:
I saw somewhere you can just use `()`, not sure if it works for a matrix, could you check? https://eigen.tuxfamily.org/dox/group__TutorialMatrixArithmetic.html

Reply:
It's the default constructor. https://eigen.tuxfamily.org/dox/classEigen_1_1Matrix.html#title7

Reply:
No argument can only be a default constructor, sorry, what I meant was can we do something like 
`Eigen::Matrix<double, 2, 2> test (1.0, 2.0, 3.0, 4.0);`

Reply:
I don't think it's the way we initialize `Eigen`. https://forum.kde.org/viewtopic.php?f=74&t=107441

Reply:
The default constructor `()` should be filled with the size of row and cols, not with the elements of the matrix. Thus I feel it's OK to use `()`. Another option is to use this:

`= (Eigen::MatrixXd rotation_matrix(2, 2) << 1.0, 2.0, 3.0, 4.0).finished()`

If not, we can always not used `const` and used comma-initialization.



Eigen::Matrix< _Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols >::Matrix | ( | Index | rows,
  |   | Index | cols
  | )




Eigen::Matrix< _Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols >::Matrix | ( | Index | rows,
  |   | Index | cols
  | )

Reply:
Yes, I know we can do `.finished()`, if you look at affine_trasform.cc, we use that. I know we can do `Eigen::Vector3d test(1.0, 2.0, 3.0)`, my question was is it only specific to vectors or does it extend to matrices?

Reply:
I tried that in matrix, I don't think it extends to matrices unfortunately. To my understanding, if we do `Eigen::Matrix<double, 3, 3> test(...)` the `(...)` is reserved to construct the size of the matrix instead of the value of elements of matrix.

Reply:
Interesting, considering `Eigen::VectorXd` is an alias for `Eigen::Matrix<double, XX, 1>`, so technically both should have the same function calls. Let's stick with `.finished()` like we use in other parts of the code.